src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -148,6 +149,10 @@ void primitive_inst::update_shape() {
         }
     }
 
+    // We assume that tensor ranks are static, thus shape_of doesn't need to update anything even if input shape is dynamic
+    if (_node.is_type<shape_of>())
```

Comment:
The impl should be selected (and built) again, even though the input layout is changed only.

Reply:
looks good to me except this.

Reply:
Fixed